portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase

```diff
@@ -0,0 +1,71 @@
+definition {
+
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Remote Apps";
+	property testray.main.component.name = "Remote Apps";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		Navigator.openURL();
+
+		if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			RemoteApps.tearDown();
+
+			Page.tearDownCP();
+		}
+	}
+
+	@priority = "5"
+	test AddRemoteApp {
+		property portal.acceptance = "true";
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Custom Apps",
+			panel = "Applications",
+			portlet = "Remote Apps");
+
+		RemoteApps.addApp(
+			name = "Test Remote App",
+			url = "http://remote-app-test.wincent.com/");
```

Comment:
This is ok for now, but in the long term we would ideally move this to a place that doesn't depend on my GitHub account. Reasons this could fail in the future:

- I go crazy and delete the corresponding repo (it's hosted on GitHub pages, so when the repo goes, the pages go).
- I forget that the repo is being used in the tests and I end up pushing something breaking into it (unlikely I know, but mentioning it for completeness).
- I get hit by a bus or for some other reason stop paying my AWS bill (the domain name's DNS is handled by Amazon's Route 53 DNS service).

Requirements for the replacement:

- Has to be on a different domain than the test instance.
- Has to be HTTP, not HTTPS (at least, it had to be HTTP during development because the portal instance runs in HTTP during development unless you jump through hoops to get a self-signed certificate working in local dev, and the schemas must match).
- That in turn means that vanilla GitHub pages sadly won't work; those are always HTTPS, so you need a custom domain.

As I said, not a blocker for this diff. But something that will break X years from now if we don't eventually move it somewhere "permanent".

Reply:
This is another example of using external artefacts out of our control for our tests. 

I would consider this non-blocking only if the next thing we do is getting rid of this and figuring out the proper test infrastructure we need for it. If there are chances we'll park this for some time, then I wouldn't be so happy with letting this through.

Reply:
ok we can take a look at getting a more robust test artifact.